repos/system_upgrade/el7toel8/libraries/rhsm.py

```diff
@@ -108,26 +113,75 @@ def get_attached_skus(context, rhsm_info):
             rhsm_info.attached_skus = _RE_SKU_CONSUMED.findall(result['stdout'])
 
 
-@with_rhsm
 def get_available_repo_uids(context, rhsm_info):
+    """Deprecated. Repositories have ids, not uids. Use get_available_repo_ids instead."""
+    return get_available_repo_ids(context, rhsm_info)
+
+
+def get_available_repo_ids(context, rhsm_info):
     """
-    Retrieves all available repositories names from the subscription manager.
+    Retrieve repo ids of all the repositories available through the subscription-manager.
 
     :param context: An instance of a mounting.IsolatedActions class
     :type context: mounting.IsolatedActions class
     :param rhsm_info: An instance of a RHSMInfo derived model.
     :type rhsm_info: RHSMInfo derived model
     """
     if not rhsm_info.available_repos:
-        with _handle_rhsm_exceptions():
-            result = context.call(['subscription-manager', 'repos'], split=False)
-            rhsm_info.available_repos = _RE_REPO_UID.findall(result['stdout'])
+        result = context.call(['yum', 'repoinfo'])
+        all_repos = list(get_repos(result['stdout']))
+        rhsm_info.available_repos = [repo.id for repo in all_repos if repo.file == '/etc/yum.repos.d/redhat.repo']
+
+
+def get_repos(repos_raw):
+    """
+    Generator providing all the repos available through yum/dnf.
+
+    :rtype: Iterator[:py:class:`leapp.libraries.common.rhsm.Repo`]
+    """
+    # TODO: Log a warning on the following msg that `yum repoinfo` may produce:
+    #  "Repository rhel-7-server-eus-rpms is listed more than once in the configuration"
+
+    # Split all the available repos per one repo
+    for repo_raw_params in re.findall(
+            r"Repo-id.*?Repo-filename.*?\n",
+            repos_raw,
+            re.DOTALL | re.MULTILINE):
+        yield parse_repo_params(repo_raw_params)
+
+
+def parse_repo_params(repo_raw_params):
+    """Parse multiline string holding repo parameters to distill the important ones."""
+    repo = Repo()
+    try:
+        repo.id = get_repo_param(r"^Repo-id\s+:\s+(.*?)(/.*)?$", repo_raw_params, "Repo-id")
+        repo.file = get_repo_param(r"^Repo-filename:\s+(.*?)$", repo_raw_params, "Repo-filename")
+    except ValueError, err:
+        api.current_logger().warn("Failed to parse the '%s' repo parameter of the `yum repoinfo` output", err.args[0])
+
+    return repo
+
+
+def get_repo_param(pattern, repo_raw_params, param):
```

Comment:
reading smoe functions, they looks they have `internal` purposes. Could you distinguish these using the `_` prefix? So it will be more clear for people, what they should/shouldn't use as it's shared library. as well, from the deprecation point of view, we will not have to care about renaming/changing such functions in future if we want.